packages/web3-eth-contract/types/index.d.ts

```diff
@@ -0,0 +1,93 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Josh Stevens <joshstevens19@hotmail.co.uk>
+ * @date 2018
+ */
+
+import {AbstractProviderAdapter, provider} from 'web3-providers';
+
+export class Contract {
+    constructor(
+        provider: AbstractProviderAdapter | provider,
+        jsonInterface: {},
+        address?: string,
+        options?: ContractOptions
+    )
+    options: Options;
+    clone(): Contract;
+    deploy(options: DeployOptions): DeployTransactionResponse;
+    methods: any;
+    once(event: string, callback: (error: Error, event: EventData) => void): void;
+    once(event: string, options: EventOptions, callback: (error: Error, event: EventData) => void): void;
+    events: any;
```

Comment:
The correct definition would be: ```events: Function[]```

Reply:
As discussed offline has to stay `any` due to not knowing what the method is being executed i.e. events.josh() in theory `josh` in compile time will be `any` not `Function[]` so will not work 👍